dvc/stage/run.py

```diff
@@ -12,27 +14,58 @@
 logger = logger.getChild(__name__)
 
 
-def _make_cmd(executable, cmd):
-    if executable is None:
-        return cmd
-    opts = {"zsh": ["--no-rcs"], "bash": ["--noprofile", "--norc"]}
-    name = os.path.basename(executable).lower()
-    return [executable, *opts.get(name, []), "-c", cmd]
-
+def _fish_supports_no_config(executable) -> bool:
+    """
+    Check if the fish shell supports the --no-config option.
 
-def warn_if_fish(executable):
-    if executable is None or os.path.basename(os.path.realpath(executable)) != "fish":
-        return
+    Parameters:
+    executable (str): The path to the fish shell executable.
 
+    Returns:
+    bool: True if the fish version is greater than 3.3.0, False otherwise.
+    """
+    try:
+        output = subprocess.check_output(
+            [executable, "-c", "echo $FISH_VERSION"],  # noqa: S603
```

Comment:
is there something like `--version`?

Reply:
there is, but extracting the version from there is more cumbersome as the output is something like `fish, version 3.7.1` where $FISH_VERSION just contains `3.7.1`

Reply:
@oswdm, `-c` will load the config, so I'll suggest using `fish --version`.

You can get the output as `output.split()[2]`, which is equivalent to what we do in the shell with the following:

```shell
fish --version | awk '{print $3}'
3.7.1
```

Please remember to catch for `IndexError`/`InvalidVersion`, but I'd expect the `--version` to remain stable.

Reply:
thanks for the suggestion @skshetry, I didn't consider the overhead of initialising a new shell.
I've changed it to `fish --version` and added checks for the exceptions in https://github.com/iterative/dvc/pull/10445/commits/b3bde770fd87a4ac889ab93d5b5c533a583e8d48